libs/media-player/src/audio-annotate.cy.tsx

```diff
@@ -0,0 +1,234 @@
+                    audioElement.play();
+                });
+            });
+
+            describe(`the mark in-point button`, () => {
+                it(`should be enabled`, () => {
+                    cy.getByDataAttribute('in-point-marker-button').should('be.enabled');
+                });
+            });
+
+            describe(`the mark out-point button`, () => {
+                it(`should be disabled`, () => {
+                    cy.getByDataAttribute('in-point-marker-button').should('be.disabled');
+                });
+            });
+
+            describe(`when the mark in-point button is clicked`, () => {
+                it(`should mark an in-point`, () => {
+                    cy.wait(2000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+                });
+            });
+
+            describe(`when the mark in-point button is clicked, the mark out-point button`, () => {
+                it('should be enabled', () => {
+                    cy.wait(2000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+
+                    cy.wait(2300);
+
+                    cy.getByDataAttribute('out-point-marker-button').should('be.enabled');
+                });
+            });
+
+            describe(`when the mark in-point button is clicked, clicking the mark out-point button`, () => {
+                it('should mark an out-point', () => {
+                    cy.wait(2000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+
+                    cy.wait(2500);
+
+                    cy.getByDataAttribute('out-point-marker-button').click();
+
+                    cy.getByDataAttribute('time-range-selected').should('exist');
+
+                    // If I remove this cy.wait() the test suite fails at the first click
+                    cy.wait(2000);
+                });
+            });
+
+            describe(`when the mark in-point button is clicked and the user scrubs the playhead back before the in-point, clicking the mark out-point button`, () => {
+                it('should fail with an error', () => {
+                    cy.wait(4000);
+
+                    cy.getByDataAttribute('in-point-marker-button').click();
+
+                    cy.get('audio').then(([audioElement]) => {
+                        audioElement.currentTime = 1.0;
+                    });
+
+                    cy.getByDataAttribute('out-point-marker-button').click();
+
+                    cy.getByDataAttribute('time-range-selected').should('not.exist');
+
+                    cy.getByDataAttribute('audio-error-message').should('exist');
+
+                    // If I remove this cy.wait() the test suite fails at the first click
+                    cy.wait(2000);
+                });
+            });
+
+            describe(`after an in-point is selected and the clear range button is clicked`, () => {
```

Comment:
Did you check that the clear selection button is disabled when there is no selection?

Reply:
With this new selected time range visualization, the clear button does not exist until at least an in point has been selected